src/native/corehost/apphost/standalone/hostfxr_resolver.cpp

```diff
@@ -8,6 +8,59 @@
 #include "trace.h"
 #include "hostfxr_resolver.h"
 
+namespace
+{
+    // SHA-256 of "dotnet-search" in UTF-8
+    #define EMBED_DOTNET_SEARCH_HI_PART_UTF8 "19ff3e9c3602ae8e841925bb461a0adb"
+    #define EMBED_DOTNET_SEARCH_LO_PART_UTF8 "064a1f1903667a5e0d87e8f608f425ac"
+
+    // <fxr_resolver::search_location_default> \0 <app_relative_dotnet_placeholder>
+    #define EMBED_DOTNET_SEARCH_FULL_UTF8    ("\0\0" EMBED_DOTNET_SEARCH_HI_PART_UTF8 EMBED_DOTNET_SEARCH_LO_PART_UTF8)
+
+    bool try_get_dotnet_search_options(fxr_resolver::search_location& out_search_location, pal::string_t& out_app_relative_dotnet)
+    {
+        constexpr int EMBED_SIZE = 512;
+        static_assert(sizeof(EMBED_DOTNET_SEARCH_FULL_UTF8) / sizeof(EMBED_DOTNET_SEARCH_FULL_UTF8[0]) < EMBED_SIZE, "Placeholder value for .NET search options longer than expected");
+
+        // Contains the EMBED_DOTNET_SEARCH_FULL_UTF8 value at compile time or app-relative .NET path written by the SDK (dotnet publish).
+        static char embed[EMBED_SIZE] = EMBED_DOTNET_SEARCH_FULL_UTF8;
+
+        out_search_location = (fxr_resolver::search_location)embed[0];
+        assert(embed[1] == 0); // NUL separates the search location and embedded .NET root value
+        bool is_configured = out_search_location != fxr_resolver::search_location_default;
+        if ((out_search_location & fxr_resolver::search_location_app_relative) != 0)
+        {
+            // Since the single static string is replaced by editing the executable, a reference string is needed to do the compare.
+            // So use two parts of the string that will be unaffected by the edit.
+            static const char hi_part[] = EMBED_DOTNET_SEARCH_HI_PART_UTF8;
+            static const char lo_part[] = EMBED_DOTNET_SEARCH_LO_PART_UTF8;
+            size_t hi_len = (sizeof(hi_part) / sizeof(hi_part[0])) - 1;
+            size_t lo_len = (sizeof(lo_part) / sizeof(lo_part[0])) - 1;
+
+            std::string binding(&embed[2]); // Embedded path is null-terminated
+            if ((binding.size() >= (hi_len + lo_len)) &&
+                binding.compare(0, hi_len, &hi_part[0]) == 0 &&
+                binding.compare(hi_len, lo_len, &lo_part[0]) == 0)
```

Comment:
```suggestion
            if ((binding.size() >= (hi_len + lo_len))
                || binding.compare(0, hi_len, &hi_part[0]) == 0
                || binding.compare(hi_len, lo_len, &lo_part[0]) == 0)
```

Shouldn't these be ORs? As in if the size is greater than the hi/lo parts that itself indicates a failure?

Reply:
I see. We should have a check for the length of the string exceeding the maximum length then.